### ISSUE TITLE
Limit size of the documentation hover?

### ISSUE THREAD

**Issue by dpc** [ul/kak-lsp#135]
Sometimes it takes the whole screen: https://i.imgur.com/4JJqHGB.png

BTW. It looks like it won't display if it can't fit.

**Comment by ul** [comment]
You can limit message size by overriding `lsp-show-hover`, I don't think it could be done in the way which fits everyone, so better to not do it by default but allow user to choose.

Also ref #75

**Comment by dpc** [comment]
I think some sensible defaults shipped by default wouldn't hurt.

Any instructions how exactly I should override `lsp-show-hover`?

**Comment by ul** [comment]
Something like this should do the job:

```
def -override -hidden lsp-show-hover -params 2 -docstring "Render hover info" %{ evaluate-commands %sh{
    # update 40 with your preference of how many lines to cut
    content=$(printf %s "$2" | head -n 40 | sed s/\'/\'\'/g)
    case $kak_opt_lsp_hover_anchor in
        true) printf "info -anchor %%arg{1} '%s'" $content;;
        *)      printf "info '%s'" $content;;
    esac
}}


```

**Comment by ul** [comment]
This command is hidden for the only purpose not to litter autocompletion as it's not supposed to be used from the prompt. But it's not private in any mean and exists solely for the possibility to override it.

But you are totally right, it worth to be documented. Care to add a wiki page?

**Comment by dpc** [comment]
Why can't it be an option? ` lsp-show-hover` would check if the option is set, and if it is, it would run the content through a function that was defined in this option. This way it's not hidden, it doesn't pollute autocompletion, and is fully plugable.

**Comment by dpc** [comment]
I've created a wiki page here: https://github.com/ul/kak-lsp/wiki/Customize-hover-documentation-size-content , but I still think that and `option` is the idiomatic way to do it.

**Comment by ul** [comment]
Yes, I tend to agree now that option worth adding. I'll work on it next week, now going offline for few days. PR is welcome as well, of course.
